src/components/dialogs/parameters/widget/ParameterLine.tsx

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { FunctionComponent, useState } from 'react';
+import { useParameterState, useStyles } from '../parameters';
+import { Grid, MenuItem, Select, Slider, Switch } from '@mui/material';
+import { FormattedMessage } from 'react-intl';
+import { Mark } from '@mui/base/useSlider';
+
+export enum ParameterType {
+    Switch,
+    DropDown,
+    Slider,
+}
+
+type BaseParameterLineProps = {
+    readonly type: ParameterType;
+    readonly param_name_id: string;
+    disabled?: boolean;
+};
+
+type SwitchParameterLineProps = {
+    readonly type: ParameterType.Switch;
+    label?: string;
+};
+
+type DropDownParameterLineProps = {
+    readonly type: ParameterType.DropDown;
+    labelTitle?: string;
+    labelValue?: string;
+    values: Record<string, string>;
+    defaultValueIfNull?: boolean;
+};
+
+type SliderParameterLineProps = {
+    readonly type: ParameterType.Slider;
+    value: number;
+    label: string;
+    marks: boolean | Mark[];
+    minValue?: number; //default = 0;
+    maxValue?: number; //default = 100;
+};
+
+type ParameterLineProps = BaseParameterLineProps &
+    (
+        | SwitchParameterLineProps
+        | DropDownParameterLineProps
+        | SliderParameterLineProps
+    );
+
+/**
+ * This component represents a parameter in the list.
+ * This is an effort to uniformize the UI style.
+ * @constructor
+ * @param props Parameter line component properties
+ * @param context React context
+ */
+export const ParamLine: FunctionComponent<ParameterLineProps> = (
```

Comment:
I think it's an antipattern to create a component that has nothing but a switch/case returning other components. Use the components directly